rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,633 @@
+                    )
+                    continue
+        return data
+
+    def _process_trove_events(
+        self,
+        changes: List[Dict[str, Any]],
+        from_timestamp: Timestamp,
+        to_timestamp: Timestamp,
+    ) -> List[DefiEvent]:
+        events = []
+        total_lusd_trove_balance = Balance()
+        realized_trove_lusd_loss = Balance()
+        for change in changes:
+            try:
+                operation = TroveOperation.deserialize(change['troveOperation'])
+                collateral_change = deserialize_asset_amount(change['collateralChange'])
+                debt_change = deserialize_asset_amount(change['debtChange'])
+                timestamp = change['transaction']['timestamp']
+                if timestamp < from_timestamp:
+                    continue
+                if timestamp > to_timestamp:
+                    break
+
+                got_asset: Optional[Asset]
+                spent_asset: Optional[Asset]
+                pnl = got_asset = got_balance = spent_asset = spent_balance = None
+                count_spent_got_cost_basis = False
+                # In one transaction is possible to generate debt and change the colateral
+                if debt_change != AssetAmount(ZERO):
+                    if debt_change > ZERO:
+                        # Generate debt
+                        count_spent_got_cost_basis = True
+                        got_asset = A_LUSD
+                        got_balance = Balance(
+                            amount=debt_change,
+                            usd_value=query_usd_price_or_use_default(
+                                A_LUSD,
+                                timestamp,
+                                ZERO,
+                                'Liquity',
+                            ),
+                        )
+                        total_lusd_trove_balance += got_balance
+                    else:  # payback debt
+                        count_spent_got_cost_basis = True
+                        spent_asset = A_LUSD
+                        spent_balance = Balance(
+                            amount=abs(debt_change),
+                            usd_value=query_usd_price_or_use_default(
+                                A_LUSD,
+                                timestamp,
+                                ZERO,
+                                'Liquity',
+                            ),
+                        )
+                        total_lusd_trove_balance -= spent_balance
+                        balance = total_lusd_trove_balance.amount + realized_trove_lusd_loss.amount
+                        if balance < ZERO:
+                            pnl_balance = total_lusd_trove_balance + realized_trove_lusd_loss
+                            realized_trove_lusd_loss += -pnl_balance
+                            pnl = [AssetBalance(asset=A_LUSD, balance=pnl_balance)]
+                if collateral_change != AssetAmount(ZERO):
```

Comment:
Helps separate the code visually when reading it

```suggestion

                if collateral_change != AssetAmount(ZERO):
```